pkg/controllers/targetversion/target_version_controller.go

```diff
@@ -7,7 +7,6 @@ import (
 	"time"
 
 	appsv1 "k8s.io/api/apps/v1"
-	corev1 "k8s.io/api/core/v1"
```

Comment:
you claim you're removing a duplicate but you don't say what is this code duplicating. I also have a feeling that it's not a duplicate until `add callback to get tls config for an endpoint` merges.

Please make this way easier to review - and add a bit of logical isolation to each change for future references - by (in terms of commit ordering):
1. add callback to get tls config for an endpoint to the generic controller and add TODO-styled tls config getters for all the endpoint check controllers
2. `remove duplicate`-commits while also adding the respective tls-config retrieving code to the controllers
2b - mention the controller that the code is duplicating in the commit messages